app.rb

```diff
@@ -0,0 +1,81 @@
+require 'rack/csrf'
+require 'sinatra'
+require 'sinatra/json'
+require 'omniauth'
+require 'omniauth-github'
+require 'sinatra/activerecord'
+require './models/user'
+require './models/task_list'
+require './models/item'
+
+secrets = YAML.load_file('secrets.yml')
+
+configure do
+  set :sessions, true
+  set :inline_templates, true
+end
+
+helpers do
+  def csrf_token
+    Rack::Csrf.csrf_token(env)
+  end
+
+  def csrf_tag
+    Rack::Csrf.csrf_tag(env)
+  end
+end
+
+use OmniAuth::Builder do
+  provider :github, secrets['github_key'], secrets['github_secret']
+end
+
+post '/add_item' do
+  if session[:authenticated]
+    item = Item.create(value:        params[:value],
+                       task_list_id: params[:task_list_id])
+    json item: { value: item.value, id: item.id }
+  else
+    status 401
+  end
+end
+
+delete '/items/:id' do
+  item = Item.find(params[:id])
+  item.destroy
+  json item_id: params[:id]
+end
+
+put '/items/:id' do
+  item = Item.find(params[:id])
+  item.update(done: !item.done)
+  json item: { done: item.done, id: item.id }
+end
+
+get '/' do
+  @user = User.find(session[:user_id]) if session[:authenticated]
+
+  erb :index
+end
+
+get '/auth/:provider/callback' do
+  auth = request.env['omniauth.auth']
+  @user = User.from_omniauth(auth)
+  if @user
+    session[:user_id] = @user.id
+    session[:user_name] = @user.name
```

Comment:
it's enough to store only user_id. Then you can implement current_user method that will fetch user from DB by id or return nil.